process/smartContract/builtInFunctions/esdtTransfer.go

```diff
@@ -20,24 +22,31 @@ var _ process.BuiltinFunction = (*esdtTransfer)(nil)
 var zero = big.NewInt(0)
 
 type esdtTransfer struct {
-	funcGasCost uint64
-	marshalizer marshal.Marshalizer
-	keyPrefix   []byte
+	funcGasCost    uint64
+	marshalizer    marshal.Marshalizer
+	keyPrefix      []byte
+	pauseHandler   process.ESDTPauseHandler
+	payableHandler process.IsPayableHandler
 }
 
 // NewESDTTransferFunc returns the esdt transfer built-in function component
 func NewESDTTransferFunc(
 	funcGasCost uint64,
 	marshalizer marshal.Marshalizer,
+	pauseHandler process.ESDTPauseHandler,
```

Comment:
payableHandler is not sent and set but is used below. If there would have been some unit tests this would have not been happened :)

Reply:
payableHandler is given via SET function. It is a chicken and egg problem in the constructors.

Reply:
done